clang/lib/AST/Mangle.cpp

```diff
@@ -302,7 +302,7 @@ void MangleContext::mangleBlock(const DeclContext *DC, const BlockDecl *BD,
     assert((isa<NamedDecl>(DC) || isa<BlockDecl>(DC)) &&
            "expected a NamedDecl or BlockDecl");
     if (isa<BlockDecl>(DC))
-      for (; DC && isa<BlockDecl>(DC); DC = DC->getParent())
+      for (; isa_and_nonnull<BlockDecl>(DC); DC = DC->getParent())
         (void) getBlockId(cast<BlockDecl>(DC), true);
```

Comment:
Could you remove the line `if (isa<BlockDecl>(DC))` as well? It's completely redundant with the loop condition.

Although I see that this is a big automatic change, so perhaps it should be left for a separate change.

Reply:
@NagyDonat thank you for the suggestion. I can remove the check after merging this PR as a separate change.